File: src/test/java/testdata/java/GenericTestPOJO.java
package testdata.java;

import java.util.ArrayList;
import java.util.List;

public class GenericTestPOJO {


    private List<Integer> list = new ArrayList<>();

    private List<Integer[]> listArr = new ArrayList<>();

    private List<List<Integer>[]> listListArr = new ArrayList<>();

    private List<Type> listEnum = new ArrayList<>();

    private List<List<Integer>> listList = new ArrayList<>();

    private List<List<List<Integer>>> listListList = new ArrayList<>();

    private List<Object> listObject = new ArrayList<>();

    private List<?> listGenericObject = new ArrayList<>();

    // --

    private Generic<Integer> generic = new Generic<>();

    private Generic<Integer[]> genericArr = new Generic<>();

    private Generic<List<Integer>[]> genericListArr = new Generic<>();

    private Generic<Type> genericEnum = new Generic<>();

    private Generic<Generic<Integer>> genericGeneric = new Generic<>();

    private Generic<Generic<Generic<Integer>>> genericGenericGeneric = new Generic<>();

    private Generic<Object> genericObject = new Generic<>();

    private Generic<?> genericGenericObject = new Generic<>();

    public class Generic<T> {

        private String username;

        private T data;
    }

    public enum Type {
        TYPE_A,
        TYPE_B,
        TYPE_C;
    }
}

File: src/test/java/testdata/kotlin/GenericTestPOJO.kt
package testdata.kotlin

class GenericTestPOJO {
    private val list: List<Int> = ArrayList()
    private val listArr: List<Array<Int>> = ArrayList()
    private val listListArr: List<Array<List<Int>>> = ArrayList()
    private val listEnum: List<Type> = ArrayList()
    private val listList: List<List<Int>> = ArrayList()
    private val listListList: List<List<List<Int>>> = ArrayList()
    private val listObject: List<Any> = ArrayList()
    private val listGenericObject: List<*> = ArrayList<Any>()

    // --
    private val generic: Generic<Int> = Generic<Int>()
    private val genericArr: Generic<Array<Int>> = Generic<Array<Int>>()
    private val genericListArr: Generic<Array<List<Int>>> = Generic<Array<List<Int>>>()
    private val genericEnum: Generic<Type> = Generic<Type>()
    private val genericGeneric: Generic<Generic<Int>> = Generic<Generic<Int>>()
    private val genericGenericGeneric: Generic<Generic<Generic<Int>>> = Generic<Generic<Generic<Int>>>()
    private val genericObject: Generic<Any> = Generic<Any>()
    private val genericGenericObject: Generic<*> = Generic<Any>()

    inner class Generic<T> {
        private val username: String? = null
        private val data: T? = null
    }

    enum class Type {
        TYPE_A, TYPE_B, TYPE_C
    }
}